pkg/vault/vault.go

```diff
@@ -0,0 +1,176 @@
+package vault
+
+import (
+	"github.com/urfave/cli"
+
+	"fmt"
+	"io/ioutil"
+	"encoding/base64"
+	"encoding/json"
+	"os"
+	"net/url"
+	"net/http"
+	"time"
+
+//	"github.com/aws/aws-sdk-go/aws"
+//	"github.com/aws/aws-sdk-go/aws/credentials"
+//	"github.com/aws/aws-sdk-go/aws/credentials"
+//	"github.com/aws/aws-sdk-go/aws/session"
+	api "github.com/hashicorp/vault/api"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/reverbdotcom/rv/pkg/iam"
+)
+
+type RoleTokens map[string]Token
+
+type Token struct {
+	Token string `json:"token"`
+	Expiration time.Time `json:"expiration"`
+}
+
+func configFileName() string {
```

Comment:
Can we put this in ~/.config or whatever is specified by the xdg spec: https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html so it plays well with other dot file / os's?